include/graaflib/properties/vertex_properties.tpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+namespace graaf::properties {
+
+template <typename V, typename E, graph_spec S>
+std::size_t get_vertex_degree(const graaf::graph<V, E, S>& graph,
+                              vertex_id_t vertex_id) {
+  if constexpr (S == graph_spec::DIRECTED) {
+    return get_vertex_outdegree(graph, vertex_id) +
+           get_vertex_indegree(graph, vertex_id);
+  }
+
+  if constexpr (S == graph_spec::UNDIRECTED) {
+    return (graph.get_neighbors(vertex_id)).size();
+  }
+
+  // Should never reach this
+  std::abort();
+}
+
+template <typename V, typename E, graph_spec S>
+std::size_t get_vertex_outdegree(const graaf::graph<V, E, S>& graph,
+                                 vertex_id_t vertex_id) {
+  return (graph.get_neighbors(vertex_id)).size();
+}
+
+template <typename V, typename E, graph_spec S>
+std::size_t get_vertex_indegree(const graaf::graph<V, E, S>& graph,
+                                vertex_id_t vertex_id) {
+  if constexpr (S == graph_spec::DIRECTED) {
+    int indegree{0};
+    for (const auto& current_vertex : graph.get_vertices()) {
+      if (graph.get_neighbors(current_vertex.first).contains(vertex_id)) {
+        indegree++;
+      }
+    }
+    return indegree;
```

Comment:
Depending on whether we expect graphs to be spart/dense it might also be worth to loop over the edges. I would propose to leave this as-is for now until we have some actual usage though. 

If you are up for it, a slightly more modern C++ experience would be to use `std::algorithm`. However I can also do this in a follow-up.

**Iterating over all vertices**
```c++
// Requires including <algorithm>
return std:ranges::count_if(graph.get_vertices(), [&graph, vertex_id](const vertex_id_to_vertex_t::value_type& kv_pair) {
    const auto& [current_vertex_id, _]{kv_pair};
    return graph.get_neighbors(current_vertex_id).contains(vertex_id);
});
```

**Iterating over all edges**
```c++
// Requires including <algorithm>
return std::ranges::count_if(graph.get_edges(), [vertex_id](const vertex_ids_to_edge_t::value_type& kv_pair) {
    const auto& [edge_id, _]{kv_pair};
    return edge_id.first == vertex_id || edge_id.second == vertex_id;
});
```

Reply:
> Depending on whether we expect graphs to be spart/dense it might also be worth to loop over the edges. I would propose to leave this as-is for now until we have some actual usage though.

I think it's hard to come up with a generic approach here. Depending on the graph architecture, the one or the other might be faster.

> If you are up for it, a slightly more modern C++ experience would be to use std::algorithm.

Damn, I feel this wasn't there in C++14. Covered in [50d78ac](https://github.com/bobluppes/graaf/pull/29/commits/50d78ac39e5fdd130a79e7750c834e47c954f256).

Reply:
Very nice, thanks!

Yes, luckily we have C++20 now 😉